### PR TITLE
Make FSharpAccessibility.IsProtected public, optimize IsPublic property

### PR DESCRIPTION
**Pull request by auduchinok**
Small changes to FSharpAccessibility in Symbols API.

### CODE REVIEW

File: src/fsharp/symbols/Symbols.fs [hunk 1]
```diff
@@ -34,7 +34,7 @@ type FSharpAccessibility(a:Accessibility, ?isProtected) =
         | _ when List.forall isInternalCompPath p  -> Internal 
         | _ -> Private
 
-    member __.IsPublic = not isProtected && match a with Public -> true | _ -> false
+    member __.IsPublic = not isProtected && match a with TAccess [] -> true | _ -> false
```
Comment (baronfel) [inline src/fsharp/symbols/Symbols.fs]: if I recall correctly, you can mark the `Public|Internal|Private` AP `internal` and the allocation gets erased entirely so you can continue to use the AP in the idiomatic way.

Comment (auduchinok) [reply src/fsharp/symbols/Symbols.fs]: The AP is type-private, isn't it the same for optimizer?

Comment (baronfel) [reply src/fsharp/symbols/Symbols.fs]: Example [sharplab](https://sharplab.io/#v2:DYLgZgzgPgLgngBwKYAIAqBBAxlpEIoC862u+KA9mJQEYBWKwAlhDALABQnwSMKTAO2YDUACigAFAK41mWKAEkBMJACcBAQ2CTVTAG4aVUAJQpRmHHgIJTxTigcoAtoawALFAhQB3JjA/2jlAoANoAuigAtAB8KNKyTFgogQ7BAPo+bkgCKAAyLDAAdGAUqlrAZmBSORkxKDCqUkimXlGxSirqWskcjijpbXG6BiopPX3cvCiqeFLAfHa9ji4w7mYWZAThpr7+Y8HxcoMNTfsotbFgWhBIQA)

Comment (auduchinok) [reply src/fsharp/symbols/Symbols.fs]: Did you mean `inline` instead of `internal`? :)

Comment (baronfel) [reply src/fsharp/symbols/Symbols.fs]: Not sure about the type-private case you mention, but if you remove the `inline` in that sharplab you see the additional choice allocations occurring. It's also a very simple example and I'm sure there's additional context I'm missing.  And yes, I do mean `inline` :D

Comment (baronfel) [reply src/fsharp/symbols/Symbols.fs]: Did more experiments and after checking the code in class members inlining results in worse code. Ignore everything I said :)

**Pull request merged**